dockstore-client/src/main/java/io/dockstore/client/cli/nested/BaseLanguageClient.java

```diff
@@ -199,6 +212,76 @@ public long launchPipeline(String entryVal, boolean localEntry, String yamlFile,
         return 0;
     }
 
+    /**
+     * Validates the locally downloaded descriptor file has the same SHA-1 checksum as the descriptor stored in the database
+     * @param type CWL or WDL or NFL
+     * @param entryVal Tool/workflow path
+     * @return void errors out if checksums do not match, provides a warning if a remote checksum is null
+     */
+    public void validateDescriptorChecksum(ToolDescriptor.TypeEnum type, String entryVal) {
+        final String[] parts = entryVal.split(":");
+        String path = parts[0];
+
+        // workflows/checkers have a special prefix for TRS endpoints
+        String ga4ghv20Path = path;
+        if (abstractEntryClient.getEntryType().toLowerCase().equals("workflow") || abstractEntryClient.getEntryType().toLowerCase().equals("checker")) {
+            ga4ghv20Path = "#workflow/" + ga4ghv20Path;
+        }
+
+        // Get the entry version we are validating for
+        String versionID = abstractEntryClient.getVersionID(entryVal);
+
+        List<ToolFile> allDescriptors = abstractEntryClient.getAllToolDescriptors(type.toString(), path, versionID);
+
+        Ga4Ghv20Api ga4ghv20api = abstractEntryClient.getClient().getGa4Ghv20Api();
+
+        // All secondary files are located relative to the location of the primary descriptor.
+        final String localTemporaryDirectory = localPrimaryDescriptorFile.getParent() + "/";
+
+        // Validate each tool file associated with the entry (Primary and secondary descriptors)
+        for (ToolFile toolFile : allDescriptors) {
+            Checksum remoteDescriptorChecksum = null;
+            Checksum localDescriptorChecksum = null;
+
+            // Get remote descriptor checksum
+            try {
+                // The TRS endpoint only discovers published entries
+                final FileWrapper remoteDescriptor = ga4ghv20api.toolsIdVersionsVersionIdTypeDescriptorRelativePathGet(type.toString(), ga4ghv20Path, versionID, toolFile.getPath());
+                List<Checksum> remoteChecksumList = remoteDescriptor.getChecksum()
+                    .stream()
+                    .filter(c -> c.getType().equals("sha1"))
+                    .collect(Collectors.toList());
+
+                remoteDescriptorChecksum = remoteChecksumList.size() > 0 ? remoteChecksumList.get(0) : null;
+            } catch (io.dockstore.openapi.client.ApiException ex) {
+                // The TRS endpoint throw openapi ApiExceptions
+                exceptionMessage(ex, "unable to locate remote descriptor " + ga4ghv20Path, ENTRY_NOT_FOUND);
+            }
+
+            // Get local descriptor checksum
+            try {
+                final File localDescriptor = new File(localTemporaryDirectory + toolFile.getPath());
```

Comment:
[Fixed](https://github.com/dockstore/dockstore-cli/blob/7f6c85a4a186819af1abeaf2f795939c1fb693b4/dockstore-client/src/main/java/io/dockstore/client/cli/nested/BaseLanguageClient.java#L263). According to documentation (and manual testing) if the child file path is an absolute path, it will be converted to a relative path, so there should be no issues.